src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -176,61 +207,214 @@ def __blob_is_not_complete(metadata):
     build_arguments = []
-    if not (arguments is None):
-        for name_value in arguments:
+    if not (build_args is None):
+        for name_value in build_args:
             name, value = name_value.split('=', 1)
             build_arguments.append(BuildArgument(name, value, False))
 
-    if not (secret_arguments is None):
-        for name_value in secret_arguments:
+    if not (secret_build_args is None):
+        for name_value in secret_build_args:
             name, value = name_value.split('=', 1)
             build_arguments.append(BuildArgument(name, value, True))
 
-    build_request = QueueBuildRequest(
-        image_name=image_name,
-        source_location=source_location,
-        build_parameters=build_parameters,
-        is_push_enabled=is_push_enabled,
-        timeout=timeout,
-        platform=platform,
-        build_arguments=build_arguments)
-
-    result = LongRunningOperation(cmd.cli_ctx)(client.queue(
-        build_request=build_request, resource_group_name=resource_group_name, registry_name=registry_name))
-
-    if isinstance(result, Build):
-        print("Successfully queued a build with build-id: {}. Starting to stream the logs...".format(result.build_id))
-        return acr_build_show_logs(cmd, client, registry_name, result.build_id, resource_group_name)
+    try:
+        build_request = QueueBuildRequest(
+            image_name=image_name,
+            source_location=source_location,
+            build_parameters=build_parameters,
+            is_push_enabled=is_push_enabled,
+            timeout=timeout,
+            platform=platform,
+            build_arguments=build_arguments)
+
+        result = LongRunningOperation(cmd.cli_ctx)(client.queue(
+            build_request=build_request, resource_group_name=resource_group_name, registry_name=registry_name))
+
+        print("Successfully queued a build with build-id: {}.".format(result.build_id))
+
+        if show_logs == 'true':
+            print("Starting to stream the logs...")
+            return acr_build_show_logs(cmd, client, registry_name, result.build_id, resource_group_name)
+    except Exception as err:
+        raise CLIError(err)
+
+
+def _check_remote_source_code(source_location):
+
+    # NOTE: acr_build doesn't support git repo url
+    if source_location.lower().startswith("git://"):
+        return source_location
+
+
+    response = requests.head(source_location)
+    if response.status_code < 400:
+        return source_location
+    else:
+        raise CLIError("'{}' doesn't exist.".format(source_location))
+
+
+def _check_image_name(image_name):
+    
+    # referenc: https://github.com/docker/distribution/tree/master/reference
+
+    if image_name == "":
+        raise CLIError("'--tag' value should not be empty.")
+    
+    tokens = image_name.split(':')
+    if(len(tokens) > 2):
+        raise CLIError("'--tag' value should be repository-name and optionally a tag in the 'repository-name:tag' format")
```

Comment:
NIT: our convention has been to just use `repository:tag`.

Reply:
fixed